tests/runtime/main.rs

```diff
@@ -179,87 +176,90 @@ fn tests(name: &str, dir_name: &str) -> Result<Vec<PathBuf>> {
     }
 
     #[cfg(feature = "c")]
-    for path in c.iter() {
-        let world_name = &resolve.worlds[world].name;
-        let out_dir = out_dir.join(format!("c-{}", world_name));
-        drop(fs::remove_dir_all(&out_dir));
-        fs::create_dir_all(&out_dir).unwrap();
-
-        let snake = world_name.replace("-", "_");
-        let mut files = Default::default();
-        let mut opts = wit_bindgen_c::Opts::default();
-        if let Some(path) = path.file_name().and_then(|s| s.to_str()) {
-            if path.contains("utf16") {
-                opts.string_encoding = wit_component::StringEncoding::UTF16;
+    if !c.is_empty() {
```

Comment:
Most of this change is from indenting this code. This is needed so that we only call `resolve_wit_dir` when there are actual non-Rust tests to run. This is because Rust now has one test where the world name cannot be determined from the contents of `dir` (the world is located in the 'inline' key of the call to `generate!`). 

The downside of this is that we're now resolving the world for each language instead of once. Let me know if you want me to refactor to avoid this duplicate work.